components/icon.js

```diff
@@ -21,6 +23,77 @@ export const LinkedIcon = ({ name, label, href }) => (
 
 )
 
+class IconListWrapperMedia extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {currentScroll: 0, isBeginning: true, isEnd: false}
+    this.scroll = this.scroll.bind(this);
+    this.list = null;
+
+    this.setListRef = element => {
+      this.list = element;
+    };
+  }
+
+  scroll(offset){
+    const element = this.list;
+    const width = element.offsetWidth;
+    const scrollWidth = element.scrollWidth;
+    const toScroll = scrollWidth - width;
+    const futurePosition = this.state.currentScroll + offset;
+    const offSetToMin = 0 + futurePosition;
+    const offSetToMax = toScroll - futurePosition;
+
+    // case where we are going to be resetting scroll, which does it as well if there are only 100px left or less
+    if(futurePosition <= 0 || offSetToMin <= 100){
+      animateScrollTo(0, {minDuration: 500, element, horizontal: true});
+      this.setState({currentScroll: 0, isBeginning: true})
+    }
+    // case where we are going to be scrolling to the end, which does it as well if there are only 100px left or less
+    else if(futurePosition >= toScroll || offSetToMax <= 100){
+      animateScrollTo(toScroll, {minDuration: 500, element, horizontal: true});
+      this.setState({currentScroll: toScroll, isEnd: true})
+    }
+    // case where we have not reached an endge
+    else{
+      animateScrollTo(futurePosition, {minDuration: 500, element, horizontal: true});
+      this.setState({currentScroll: futurePosition, isBeginning: false, isEnd: false})
+    }
+  }
+
+  render() {
+    const isMobile = this.props.mobile;
+    const wrapperClass = isMobile ? "" : "IconListWrapperMedia";
+    const listElementClass = isMobile ? "IconList" : "IconList IconListWrapperMedia__list";
```

Comment:
Didn't feel like using `classnames` for this one? 😄 it's a good util, has minimal footprint so don't be shy to use it.